### DISCUSSION TITLE
Is there a way to get a list of available rooms live?

### DISCUSSION THREAD

**Discussion by bedryck** [colyseus/colyseus#694] [Q&A] [answered]
Is there a way to get the list of available rooms live without calling getAvailableRooms at some interval?

I did not find anything in the documentation(

My other thoughts were to get a socket instance from Colyseus and send a socket message when the room is created and deleted

 Example: 
```
 onCreate(){
 ...
  ws.send('available-rooms', <ArrayOfAvailableRooms>);
...
 }
```
But I get such error `Error: seat reservation expired.` when I try to connect to the socket using general methods instead of Colyseus.

So the next question is, can I just connect to the socket without using Colyseus client?

**Comment by bedryck** [comment]
I have one more crazy idea. XD.  I can create a separate room that will be responsible for the available rooms. This room has a state with available rooms. Other rooms change the state of this specific room. And client has a sync callback when rooms change
Do you think this is a very bad idea?)

**Reply by endel** [reply, marked as answer]
HI @bedryck, that is essentially what the `LobbyRoom` does :) though, it doesn't use the state - it only sends data as the rooms become available or unavailable. You can take a look at the sources here: https://github.com/colyseus/colyseus/blob/master/packages/core/src/rooms/LobbyRoom.ts

The `LobbyRoom` uses the (yet undocumented) `subscribeLobby` method, which is a "global" subscription for when rooms are available/unavailable.

**Reply by bedryck** [reply]
Oh, thanks!
Sorry for missing this in the documentation and wasting your time(
